controllers/autoscaling.go

```diff
@@ -52,18 +52,37 @@ func (r *HorizontalRunnerAutoscalerReconciler) determineDesiredReplicas(rd v1alp
 			return nil, err
 		}
 
-		for _, r := range list.WorkflowRuns {
+		for _, run := range list.WorkflowRuns {
 			total++
 
 			// In May 2020, there are only 3 statuses.
 			// Follow the below links for more details:
 			// - https://developer.github.com/v3/actions/workflow-runs/#list-repository-workflow-runs
 			// - https://developer.github.com/v3/checks/runs/#create-a-check-run
-			switch r.GetStatus() {
+			switch run.GetStatus() {
 			case "completed":
 				completed++
 			case "in_progress":
-				inProgress++
+				jobs, _, err := r.GitHubClient.Actions.ListWorkflowJobs(context.TODO(), user, repoName, run.GetID(), nil)
+				if err != nil {
+					r.Log.Error(err, "Error listing workflow jobs")
+					inProgress++
```

Comment:
If I understand this correctly, this means that if a workflow has 4 jobs and all 4 are running / in progress, ``inProgress` will actually be set to 5 - 1 for the whole workflow and 4 for the child jobs?

Or am I misunderstanding it?

Reply:
No, `inProgress` will only be 4. We do not count the workflow itself when we successfully list the wf's jobs. The only time workflows are represented in the `inProgress` variable is when an error is encountered while trying to list the child jobs, or when the wf has no child jobs for some reason.